app/vmagent/remotewrite/remotewrite.go

```diff
@@ -510,15 +515,31 @@ func newRemoteWriteCtx(argIdx int, at *auth.Token, remoteWriteURL *url.URL, maxI
 	}
 
 	// Initialize sas
-	sasFile := streamAggrConfig.GetOptionalArg(argIdx)
-	if sasFile != "" {
+	sacfg := saConfigLoader.GetCurrentConfig(argIdx)
+	if len(sacfg) > 0 {
+		sasFile := streamAggrConfig.GetOptionalArg(argIdx)
 		dedupInterval := streamAggrDedupInterval.GetOptionalArgOrDefault(argIdx, 0)
-		sas, err := streamaggr.LoadFromFile(sasFile, rwctx.pushInternal, dedupInterval)
+		sas, err := streamaggr.NewAggregators(sacfg, rwctx.pushInternal, dedupInterval)
 		if err != nil {
 			logger.Fatalf("cannot initialize stream aggregators from -remoteWrite.streamAggrFile=%q: %s", sasFile, err)
 		}
 		rwctx.sas = sas
 		rwctx.streamAggrKeepInput = streamAggrKeepInput.GetOptionalArg(argIdx)
+
+		configReloaderWG.Add(1)
+		go func() {
+			defer configReloaderWG.Done()
+			for {
+				select {
+				case <-stopCh:
+					return
+				case c := <-saConfigLoader.UpdatesCh(argIdx): // wait for updates after config reload
```

Comment:
I think, dymanic updates is not needed.

 `remoteWrite.streamAggr.config` - always has the same number of files and it cannot be changed dynamically.